api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -68,23 +69,58 @@ public interface Snapshot extends Serializable {
    * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
    *
    * @return a list of ManifestFile
+   * @deprecated since 1.0.0 - Use {@link Snapshot#allManifests(FileIO)} instead.
```

Comment:
Instead of saying when it was deprecated, I think we should say when it will be removed. I'm thinking that will probably be 1.0.0, assuming that we do an 0.14.0 and then follow up with a 1.0.0 to remove deprecations, but open to other opinions.

Reply:
Yeah I think that's a good idea. I believe Yufei did something similar (noting when it would be deprecated) when deprecating the old metadata write directory configurations.

I also think that this gives a little bit more of an argument for a 0.14.0. We can go through, determine what needs to be deprecated, cut a release, and then turn around and cut a 1.0.0 release that from then on has stronger API / ABI guarantees.

Reply:
Copied existing language and mentioned it's deprecated since 0.14.0 and will be deprecated in 1.0.0.

I found a deprecation notice saying something would be removed in 0.15.0, so I think some ambiguity about 0.14.0 vs 1.0.0 is fine for now.

Reply:
Also I’ll make a note of the 0.15.0 thing and we’ll get that cleared up. Found it basically git grep-Ing and it’s very unlikely we’ll have a 0.15.0.